color_channel_swapper.py

```
@@ -0,0 +1,95 @@
+        self.save_btn.config(state=tk.NORMAL)
+
+    # Add a trace to the channel selection to disable the swap button if identical
+    def update_swap_btn_state(self, *args):
+        c1 = self.channel1.get()
+        c2 = self.channel2.get()
+        self.swap_btn.config(state=tk.NORMAL if c1 != c2 and self.image else tk.DISABLED)
+
+    def save_image(self):
+        if not self.image:
+            return
+        save_path = filedialog.asksaveasfilename(defaultextension=".png", filetypes=[("PNG files", "*.png"), ("JPEG files", "*.jpg;*.jpeg")])
```

Pylint (3.3.7)

[convention] Line too long (141/100)

(C0301)